engine/kv_engine.cpp

```diff
@@ -1119,30 +1303,34 @@ Status KVEngine::BatchWrite(const WriteBatch &write_batch) {
     return s;
   }
 
+  RAIICaller batch_write_holder(
```

Comment:
Make this a lock and guard it.

Reply:
check by try_lock() + unlock() is too heavy